CI failures, the override system seems to have made it harder to change signatures. 
```
++python -c 'import os; import numpy; print(os.path.dirname(numpy.__file__))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/numpy/__init__.py", line 145, in <module>
    from . import lib
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/numpy/lib/__init__.py", line 27, in <module>
    from .arraypad import *
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/numpy/lib/arraypad.py", line 965, in <module>
    def pad(array, pad_width, mode='constant', **kwargs):
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/numpy/core/overrides.py", line 142, in decorator
    verify_matching_signatures(implementation, dispatcher)
  File "/home/travis/build/numpy/numpy/builds/venv/lib/python3.6/site-packages/numpy/core/overrides.py", line 81, in verify_matching_signatures
    'different function signatures' % implementation)
RuntimeError: implementation and dispatcher for <function pad at 0x7f29c94da9d8> have different function signatures
```

This needs a change to 
```
def _pad_dispatcher(array, pad_width, mode, **kwargs):
    return (array,)
```
right above the `pad` function definition.

--- a/numpy/lib/arraypad.py
+++ b/numpy/lib/arraypad.py
@@ -957,7 +957,7 @@ def _as_pairs(x, ndim, as_index=False):
 # Public functions
 
 
-def _pad_dispatcher(array, pad_width, mode, **kwargs):
+def _pad_dispatcher(array, pad_width, mode=None, **kwargs):
     return (array,)
 
 

--- a/numpy/lib/tests/test_arraypad.py
+++ b/numpy/lib/tests/test_arraypad.py
@@ -1251,7 +1251,7 @@ def test_kwargs(mode):
 
 def test_constant_zero_default():
     arr = np.array([1, 1])
-    assert_array_equal(pad(arr, 2), [0, 0, 1, 1, 0, 0])
+    assert_array_equal(np.pad(arr, 2), [0, 0, 1, 1, 0, 0])
 
 
 @pytest.mark.parametrize("mode", _all_modes.keys())

